File: 1869. Longer Contiguous Segments of Ones than Zeros/solution.py
class Solution(object):
    def checkZeroOnes(self, s):
        oneSum = 0
        zeroSum = 0
        tempSum = 0
        for index, character in enumerate(s):
            if(int(character) != int(s[index-1])):
                if(s[index-1] == "1"):
                    oneSum = max(oneSum, tempSum)
                else:
                    zeroSum = max(zeroSum, tempSum)
                tempSum = 0
            tempSum += 1

        if(s[index] == "1"):
            oneSum = max(oneSum, tempSum)
        else:
            zeroSum = max(zeroSum, tempSum)
        return(oneSum > zeroSum)

File: 1869. Longer Contiguous Segments of Ones than Zeros/solution.java
class Solution {
    public boolean checkZeroOnes(String s) {
        int oneSum = 0;
        int zeroSum = 0;
        int tempSumOne = 0;
        int tempSumZero = 0;
        for(char character : s.toCharArray()){
            if(character=='1'){
                oneSum = Math.max(oneSum, ++tempSumOne);
                tempSumZero = 0;
            }
            else{
                zeroSum = Math.max(zeroSum, ++tempSumZero);
                tempSumOne = 0;
            }
        }
        return(oneSum > zeroSum);
    }
}